connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -242,6 +243,17 @@ class SparkConnectPlanner(session: SparkSession) {
       .logicalPlan
   }
 
+  private def transformWithColumns(rel: proto.WithColumns): LogicalPlan = {
+    val (names, cols) =
+      rel.getNameExprListList.asScala
+        .map(e => (e.getName(0), Column(transformExpression(e.getExpr))))
```

Comment:
should we assert on the number of parts being present in the names list?

Reply:
Documented this into proto and clients implementation can take care of it.

Reply:
I think it's better to have server-side validation if it's not hard. Or we can add a new proto message `MultiAlias` so that this is "type-safe".

Reply:
I really want to not validation more on server side. Partially because I feel like that is not good user experience (partially of the partially is because how we throw exceptions in client side).

If we ever can eliminate ambiguity (thus lead to a good API design) then why not, which can be achieved by have two proto message, one is for single alias and another is for multiple alias.

Even we include a `bool` to indicate if this is a singe alias or multi alias is better than server side validation.

Reply:
Let's not use the current crude way of doing error handling as an excuse to not do server side validation.

Reply:
The current API design is not good because it carries ambiguity thus it asks server to validate. This happens because we put two things on the same proto field then server side needs to either if else or validate one case. This is the primary point. The current way for error handling only makes it looks even terrible.

Reply:
I will close this comment by doing a validation on the server side.

Reply:
Added a server side validation with a test case.